### ISSUE TITLE
[STORM-1369] Add trident MapState to Cassandra

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#5204]
No Description

---
<details><summary><i>Originally reported by <a title="Satish Duggana" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=satish.duggana">satish.duggana</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-1369" target="_blank">Add trident MapState to Cassandra</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Magnus Koch" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mkoch1">mkoch1</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-04-05T01:50:19+01:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mkoch1">mkoch1</a>:</i>
<p>Picked this up, because we need it internally.</p>

<p>Implementation is available in pull requests <a href="https://github.com/apache/storm/pull/1781" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/1781</a> (for master) and <a href="https://github.com/apache/storm/pull/1782" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/1782</a> (for 1.0.x-branch).</p>

<p>The README shows the basic usage of it, and there is a unit test that implements the word count example for opaque, transactional and nontransactional map states.</p>

<p>The approach was to reuse much of the existing features (e.g. CQL tuple mappers) and only adding a few things to wire them together with an IBackingMap. </p>

<p>Notably:</p>
<ul>
	<li>There is now a StateMapper which converts the map state (which is generic and can be opaque/transactional) to/from a tuple format.</li>
	<li>There is a new async execution option with an AsyncResultSetHandler to collect the results - this is used by the MapState implementation which by default queries Cassandra for multiple keys in parallel.</li>
	<li>There is a builder class to simplify construction, which builds the CQL statements for you. This is not strictly necessary but it really does help usage and readability for most cases.</li>
</ul>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>:</i>
<p>Thanks <a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mkoch1" class="user-hover" rel="mkoch1">Magnus Koch</a>, I merged into master.</p>

<p>I just merged this for only master since we didn't discuss a plan (at least right now) for minor version of 1.x, and works for 2.0.0 is in progress. Please let me know or post request to dev@ when you really want to have this for 1.2.0.</p>
